data-prepper-plugins/service-map-stateful/src/main/java/com/amazon/dataprepper/plugins/prepper/ServiceMapStatefulPrepper.java

```diff
@@ -127,41 +128,37 @@ private static File createPath(File path) {
-        records.forEach(i -> i.getData().getResourceSpansList().forEach(resourceSpans -> {
-            OTelHelper.getServiceName(resourceSpans.getResource()).ifPresent(serviceName -> resourceSpans.getInstrumentationLibrarySpansList().forEach(
-                    instrumentationLibrarySpans -> {
-                        instrumentationLibrarySpans.getSpansList().forEach(
-                                span -> {
-                                    if (OTelHelper.checkValidSpan(span)) {
-                                        try {
-                                            batchStateData.put(
-                                                    span.getSpanId().toByteArray(),
-                                                    new ServiceMapStateData(
-                                                            serviceName,
-                                                            span.getParentSpanId().isEmpty() ? null : span.getParentSpanId().toByteArray(),
-                                                            span.getTraceId().toByteArray(),
-                                                            span.getKind().name(),
-                                                            span.getName()));
-                                        } catch (RuntimeException e) {
-                                            LOG.error("Caught exception trying to put service map state data into batch", e);
-                                        }
-                                        if (span.getParentSpanId().isEmpty()) {
-                                            try {
-                                                currentTraceGroupWindow.put(span.getTraceId().toByteArray(), span.getName());
-                                            } catch (RuntimeException e) {
-                                                LOG.error("Caught exception trying to put trace group name", e);
-                                            }
-                                        }
-                                    } else {
-                                        LOG.warn("Invalid span received");
-                                    }
-                                });
+        records.forEach(record -> {
+            final Span span = record.getData();
+            if (span.getServiceName() != null) {
+                final String serviceName = span.getServiceName();
+                final String spanId = span.getSpanId();
+                final String traceId = span.getTraceId();
+                final String parentSpanId = span.getParentSpanId();
+                try {
+                    batchStateData.put(
+                            Hex.decodeHex(spanId),
+                            new ServiceMapStateData(
+                                    serviceName,
+                                    parentSpanId.isEmpty()? null : Hex.decodeHex(parentSpanId),
```

Comment:
Will this `Hex.decodeHex(parentSpanId)` result in the same output as `span.getParentSpanId().toByteArray()`

Reply:
No. the decoding algorithm is different. Here is a snippet to verify

```
final String spanId = "aedf42";
        io.opentelemetry.proto.trace.v1.Span sp = io.opentelemetry.proto.trace.v1.Span.newBuilder().setSpanId(ByteString.copyFrom(spanId.getBytes())).build();
        System.out.println(Arrays.equals(sp.getSpanId().toByteArray(), spanId.getBytes()));
        System.out.println(Arrays.equals(sp.getSpanId().toByteArray(), Hex.decodeHex(spanId)));
```
First printout is true while second is false.

Reply:
Discussed offline: Will open an issue to track enhancement of

```
public class MapDbPrepperState<V> implements PrepperState<byte[], V>
```

so that we no longer need to convert spanID string back into hex decoded bytes.